cms/djangoapps/contentstore/xblock_storage_handlers/view_handlers.py

```diff
@@ -799,115 +751,15 @@ def _duplicate_block(
     store = modulestore()
     with store.bulk_operations(duplicate_source_usage_key.course_key):
         source_item = store.get_item(duplicate_source_usage_key)
-        # Change the blockID to be unique.
-        dest_usage_key = source_item.location.replace(name=uuid4().hex)
-        category = dest_usage_key.block_type
-
-        # Update the display name to indicate this is a duplicate (unless display name provided).
-        # Can't use own_metadata(), b/c it converts data for JSON serialization -
-        # not suitable for setting metadata of the new block
-        duplicate_metadata = {}
-        for field in source_item.fields.values():
-            if field.scope == Scope.settings and field.is_set_on(source_item):
-                duplicate_metadata[field.name] = field.read_from(source_item)
-
-        if is_child:
-            display_name = (
-                display_name or source_item.display_name or source_item.category
-            )
-
-        if display_name is not None:
-            duplicate_metadata["display_name"] = display_name
-        else:
-            if source_item.display_name is None:
-                duplicate_metadata["display_name"] = _("Duplicate of {0}").format(
-                    source_item.category
-                )
-            else:
-                duplicate_metadata["display_name"] = _("Duplicate of '{0}'").format(
-                    source_item.display_name
-                )
-
-        asides_to_create = []
-        for aside in source_item.runtime.get_asides(source_item):
-            for field in aside.fields.values():
-                if field.scope in (
-                    Scope.settings,
-                    Scope.content,
-                ) and field.is_set_on(aside):
-                    asides_to_create.append(aside)
-                    break
-
-        for aside in asides_to_create:
-            for field in aside.fields.values():
-                if field.scope not in (
-                    Scope.settings,
-                    Scope.content,
-                ):
-                    field.delete_from(aside)
```

Comment:
Don't we need this code anymore?

Reply:
Actually no, it's the same code as in [gather_block_attributes](https://github.com/openedx/edx-platform/blob/619958821dc2b9c965aa6170123f52520ced63be/cms/djangoapps/contentstore/utils.py#L1081).